runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -168,14 +168,13 @@ class MM_ObjectAllocationAPI
 		if (0 != size) {
 			/* Contiguous Array */
 
-			UDATA scale = ((J9ROMArrayClass*)(arrayClass->romClass))->arrayShape;
 
 #if !defined(J9VM_ENV_DATA64)
-			if (!sizeCheck || (size < ((U_32)J9_MAXIMUM_INDEXABLE_DATA_SIZE >> scale)))
+			if (!sizeCheck || (size < ((U_32)J9_MAXIMUM_INDEXABLE_DATA_SIZE >> (((J9ROMArrayClass*)(arrayClass->romClass))->arrayShape))))
 #endif /* J9VM_ENV_DATA64 */
 			{
 				/* Calculate the size of the object */
-				UDATA dataSize = ((UDATA)size) << scale;
+				UDATA dataSize = (((UDATA)size) * (J9ARRAYCLASS_GET_STRIDE(arrayClass)));
```

Comment:
stride should be in a local since you read it twice.

Reply:
Not sure what I was seeing here.

Reply:
brackets are not needed around `J9ARRAYCLASS_GET_STRIDE(arrayClass)`